### ISSUE TITLE
Suggestion: Add a "Why" section to the Introduction

### ISSUE THREAD

**Issue by neokio** [vuejs/vuejs.org#811]
I've been a full-stack LAMP guy for more than 10 years. For the last 2 years, I've repeatedly heard "Check out Vue!", and have spent maybe an hour in total reading the guides and studying the examples on vue.org. And every time, I think the same thing: "Ok, another Javascript UI library, like Angular or React. How could this help anyone? Nothing here is more efficient than PHP+MySQL. What am I not seeing? WHY do people use Javascript UI libraries?"

What makes this ...

```html
<div id="app">
  <p>{{ message }}</p>
</div>
```
```javascript
var app = new Vue({
  el: '#app',
  data: {
    message: 'Hello Vue!'
  }
})
```
... better than this ...

```php
<?php
  $message = 'Hello PHP';
?>
<div id="app">
  <p><?= $message =></p>
</div>
```
?

Maybe it's not up to Vue.org to educate people on what makes a Javascript UI library useful. Unfortunately, to the best of my searching, no one else has taken this on to any effect.

The Vue intro guide is written from the perspective of someone who already uses a Javascript UI library. There's no context, no sense of what problems MVVM frameworks solve, comparing the benefits versus server-side approaches, etc.

My suggestions is to expand the scope of the Introduction by including a section that covers the "Why", written as one would describe color to a blind person. Otherwise, you're just preaching to the converted.

**Comment by LinusBorg** [comment]
> Maybe it's not up to Vue.org to educate people on what makes a Javascript UI library useful. 

Agreed.

> Unfortunately, to the best of my searching, no one else has taken this on to any effect.

https://www.google.de/search?q=advantage+of+client-side+rendering&oq=advantage+of+client-side+rendering&aqs=chrome..69i57j0.8143j0j7&sourceid=chrome&ie=UTF-8

> There's no context, no sense of what problems MVVM frameworks solve, comparing the benefits versus server-side approaches, etc.

Apart from the point that Vue is not a MVVM framework (it's less than that), I have hard time believing that you can't the the advantage of rendering UI changes (like switching tabs) locally versus rendering a completely new version of the page on the server, where 95% of the HTML will be the same again:

It's **much, much** faster.

There's pretty much not very much other than that, but that point is **huge**. You simply **can't** write a website like facebook with only server-side rendering and have the same user experience. Imagine every "like", every adding of a comment etc would trigger a page reload!

**Comment by neokio** [comment]
Thanks for that link, and the thoughtful response!
You've made clear to me that I've asked the wrong question, and poorly.

Everything I code is laced with jQuery Ajax updating the DOM with PHP queried MySQL.
I haven't seen anything in a Javascript UI framework that I can't do more efficiently with PHP+Ajax.
I'm just looking for evidence that there's a better way.

**Comment by LinusBorg** [comment]
> I haven't seen anything in a Javascript UI framework that I can't do more efficiently with PHP+Ajax.

Well, all those frameworks think they can do that. Have you actually tried them (in more than a simple codepen), or just judged them by their docs?

**Comment by znck** [comment]
Go listen to Gitlab - https://youtu.be/ioogrvs2Ejc
